internal/webhook/stage/webhook.go

```diff
@@ -92,8 +99,13 @@ func (w *webhook) Default(ctx context.Context, obj runtime.Object) error {
 	}
 
 	var oldStage *kargoapi.Stage
-	if req.OldObject.Object != nil {
-		oldStage = req.OldObject.Object.(*kargoapi.Stage) // nolint: forcetypeassert
+	// We need to decode old object manually since controller-runtime doesn't decode it for us.
+	if req.Operation == admissionv1.Update ||
+		req.Operation == admissionv1.Delete {
```

Comment:
```suggestion
	if req.Operation == admissionv1.Update {
```

would probably be better, as we do not need the decoded object on delete.